src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
This function "consumes" value on success. Skipping the value is a success. To handle it specially would require passing the information back to parse_env_file_internal that it's not really a success, so it could free `value`. This would be more core and more brittleness. Current cure *is* quite simple, I just forgot to take care of `value` in this one case.